hasktorch/src/Torch/NN.hs

```diff
@@ -139,3 +139,35 @@ instance Parameterized Linear
 --     return $ Linear{..}
 
 instance Parameterized [Linear]
+
+data Conv2dSpec = 
+  Conv2dSpec {
+    inputChannelSize  :: Int, 
+    outputChannelSize :: Int, 
+    kernelSize0       :: Int, 
+    kernelSize1       :: Int
```

Comment:
Maybe instead of `0` and `1` suffixes call them `kernelHeight` and `kernelWidth`?